### PR TITLE
Fuzz: part 2

### PR DESCRIPTION
**Pull request by ernado**
Fuzzing for a couple of hours discovered some more bugs.

### CODE REVIEW

**Comment by Sean-Der** [comment]
LGTM! Mind squashing these down (just so crashers/fix are in the same commit)

thanks

**Comment by ernado** [comment]
I've squashed into one commit.
Thought that it will be easier to read the changes if there are code commit + corpus commit.

File: handshake_message_server_key_exchange.go [hunk 1]
```diff
@@ -53,23 +53,29 @@ func (h *handshakeMessageServerKeyExchange) Unmarshal(data []byte) error {
 
 	publicKeyLength := int(data[3])
 	offset := 4 + publicKeyLength
-	if len(data) <= publicKeyLength {
```
Comment (Sean-Der) [inline handshake_message_server_key_exchange.go]: I approved (so feel free to merge away!)

Style wise do you think it is better to do one assert for total length `if len(data) <= handshakeMessageServerKeyExchangeLen` or small asserts before each data access?

Comment (ernado) [reply handshake_message_server_key_exchange.go]: I think it is better to check as early as we can.
Probably we can use something like https://godoc.org/golang.org/x/crypto/cryptobyte to decrease probability of error like TLS library does.

**Pull request merged**